great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -47,77 +41,61 @@ def __init__(
-            data_reference=data_reference, data_asset_name=None
-        )
-        self._data_references_cache = {data_reference: mapped_batch_definition_list}
+        self._data_references_cache = {}
 
     def _get_data_reference_list(
         self, data_asset_name: Optional[str] = None
     ) -> List[str]:
-        """List objects in the underlying data store to create a list of data_references.
-
-        This method is used to refresh the cache.
         """
-        return [""]
+        List objects in the cache to create a list of data_references. If data_asset_name is passed in, method will
+        return all data_references for the named data_asset. If no data_asset_name is passed in, will return a list of
+        all data_references for all data_assets in the cache.
+        """
+        if data_asset_name:
+            return self._get_data_reference_list_from_cache_by_data_asset_name(
+                data_asset_name
+            )
+        else:
+            data_reference_list = [
+                self._get_data_reference_list_from_cache_by_data_asset_name(
+                    data_asset_name
+                )
+                for data_asset_name in self.get_available_data_asset_names()
+            ]
+            return data_reference_list
 
     def _get_data_reference_list_from_cache_by_data_asset_name(
         self, data_asset_name: str
     ) -> List[str]:
         """Fetch data_references corresponding to data_asset_name from the cache."""
-        batch_definition_list: List[
-            BatchDefinition
-        ] = self._get_batch_definition_list_from_batch_request(
-            batch_request=BatchRequestBase(
-                datasource_name=self.datasource_name,
-                data_connector_name=self.name,
-                data_asset_name=data_asset_name,
-            )
+        data_references_for_data_asset_name = self._data_references_cache.get(
+            data_asset_name
         )
-
-        if len(batch_definition_list) == 0:
-            return []
-        return [
-            self._map_batch_definition_to_data_reference(
-                batch_definition=batch_definition,
-            )
-            for batch_definition in batch_definition_list
-        ]
+        if data_references_for_data_asset_name is not None:
+            return list(data_references_for_data_asset_name.keys())
+        else:
+            return [""]
 
     def get_data_reference_list_count(self) -> int:
-        return len(self._data_references_cache)
+        """
+        Get number of data_references corresponding to all data_asset_names in cache. In cases where the
+        RuntimeDataConnector has been passed a BatchRequest with the same data_asset_name but different
+        batch_identifiers, it is possible to have more than one data_reference for a data_asset.
+        """
+        sums = 0
+        for key, data_reference_dict in self._data_references_cache.items():
```

Comment:
@Shinnnyshinshin : This summation could be performed in a single line by using the `Python` `sum()` method.

Reply:
great suggestion. Updated

Reply:
> ...Wondering if we can do a test of setting a `data_asset_name` and then seeing it returned by `get_available_data_asset_names()` from the `RuntimeDataConnector`.

- You're right, and this behavior is tested by 2 tests in `test_runtime_data_connector.py`. 
- `test_get_available_data_asset_names()` will return an empty list, since it's before any  `BatchRequest` has been passed in. 
- `test_get_available_data_asset_names_updating_after_batch_request()` will show the `data_asset_names` being set after being passed in 2 `BatchRequests`